File: leetcode/_209.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int result = int.MaxValue;
        int sum = 0;
        int j = 0;
        for (int i = 0; i < nums.Length; i++){
            sum += nums[i];
            while(sum>=target){
                int subLength = i-j+1;
                result = Math.Min(result, subLength);
                sum -= nums[j];
                j++;
            }
        }

        return result == int.MaxValue? 0:result;
    }
}

File: leetcode/_209.cpp
#include <vector>
#include <climits>

using namespace std;
class Solution
{
public:
    int minSubArrayLen(int target, vector<int> &nums)
    {
        int left = 0;
        int n = nums.size();
        int sum = 0;
        int result = INT32_MAX;
        int subLength = 0;
        for (int right = 0; right < n; right++)
        {
            sum += nums[right];
            while (sum >= target)
            {
                subLength = right - left + 1;
                result = result < subLength ? result : subLength;
                sum -= nums[left++];
            }
        }

        return result == INT32_MAX ? 0 : result;
    }
};